contents/docs/libraries/ios/configuration.mdx

```diff
@@ -0,0 +1,388 @@
+    if let email = event.properties["email"] as? String {
+        event.properties["email"] = email.map { _ in "*" }.joined()
+    }
+    
+    return event
+}
+```
+
+</details>
+
+<details>
+<summary>Drop events by event name</summary>
+
+```swift
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_api_client_host>")
+
+config.setBeforeSend { event in
+    // Drop all events named "Stale Event"
+    if event.event == "Stale Event" {
+        return nil
+    }
+    return event
+}
+```
+
+</details>
+
+### Sampling events
+
+Sampling lets you choose to send only a percentage of events to PostHog. It is a good way to control your costs without having to completely turn off features of the SDK.
+
+<details>
+<summary>Sample events by event name</summary>
+
+```swift
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_api_client_host>")
+
+config.setBeforeSend { event in
+    // Sample 10% of Sampled Event events
+    if event.event == "Sampled Event" {
+        if Double.random(in: 0...1) < 0.1 {
+            event.properties["$sample_type"] = ["sampleByEvent"]
+            event.properties["$sample_threshold"] = 0.1
+            event.properties["$sampled_events"] = ["Sampled Event"]
+            return event
+        }
+        return nil
+    }
+    return event
+}
+```
+
+</details>
+
+### Chaining multiple BeforeSendBlocks
+
+You can provide an array of `BeforeSendBlock` functions to be called one after the other:
+
+```swift
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_api_client_host>")
+
+config.setBeforeSend(
+    // First block: Drop all events named "Stale Event"
+    { event in
+        if event.event == "Stale Event" {
+            return nil
+        }
+        return event
+    },
+    // Second block: Redact sensitive information
+    { event in
+        if let email = event.properties["email"] as? String {
+            event.properties["email"] = email.map { _ in "*" }.joined()
+        }
+        return event
+    }
+)
+```
+
+**Note:** When chaining beforeSend blocks, order is important. The first block is executed first and the mutated event is passed along to the second block, and so on. If at any point in the chain the event is dropped, any subsequent blocks will not be executed.
+
+## Opt out of data capture
+
+You can completely opt-out users from data capture. To do this, there are two options:
+
+1. Opt users out by default by setting `optOut` to `true` in your PostHog config:
+
+```swift
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_client_api_host>")
+config.optOut = true
+PostHogSDK.shared.setup(config)
+```
+
+2. Opt users out on a per-person basis by calling `optOut()`:
+
+```swift
+PostHogSDK.shared.optOut()
+```
+
+Similarly, you can opt users in:
+
+```swift
+PostHogSDK.shared.optIn()
+```
+
+To check if a user is opted out:
+
+```swift
+PostHogSDK.shared.isOptOut()
+```
+
+## Method swizzling
```

Comment:
Do you think this one should go in installation too?

Maybe there is a "features" section of installation that can include it?